CONTRIBUTING.md

```diff
@@ -126,3 +126,49 @@ We encourage contributors to leverage Large Language Models (LLMs) responsibly:
   ```sh
   sbt "scalafmtCheckAll"
   ```
+
+## Design Guidelines
+
+### Adding a New Method
+
+If you want to contribute a new method like `S.newMethod` or `s.newMethod`, feel free to:
+
+* Open an issue
+    
+* Discuss on Discord: [https://discord.gg/afch62vKaW](https://discord.gg/afch62vKaW)
+    
+* Share design examples:
+    
+    * Use cases
+        
+    * Equivalent in `ZIO` or `Cats Effect`
+        
+    * Other motivating patterns
+        
+
+### Where to add Your `newMethod`
+
+| Subproject | Use For |
+| --- | --- |
+| `kyo-data` | Data structures (`Chunk`, `Maybe`, `Result`, etc.) |
+| `kyo-prelude` | Effect types without `IO` (`Abort`, `Env`, `Var`, etc.) |
+| `kyo-core` | Methods requiring `IO`, `Async`, or `Resource` |
+| `kyo-combinators` | Extensions or composition helpers |
+
+Add corresponding tests in the same subproject.
+
+**Example:**  
+A new `Stream.fromSomething` method:
+
+* If it uses `IO`: place it in `kyo-core/shared/src/main/scala/kyo/StreamCoreExtensions.scala`
+    
+* If it doesn't: place it in `kyo-prelude/shared/src/main/scala/kyo/Stream.scala`
+    
+
+### Tips
+
+* Prefer [`call-by-name`](https://docs.scala-lang.org/tour/by-name-parameters.html) (`body: => A < S`) for deferred evaluation, especially when lifting to `IO`.
```

Comment:
This works well for `IO` but not other effect suspensions since we can't leverage any suspension to defer side effects. `IO` is special in this regard since it's the last effect to be handled. Let's make the tip more specific.